apiserver/facades/controller/agenttools/agenttools.go

```diff
@@ -63,13 +66,16 @@ type newEnvironFunc func() (environs.Environ, error)
 type toolsFinder func(context.Context, tools.SimplestreamsFetcher, environs.BootstrapEnviron, int, int, []string, coretools.Filter) (coretools.List, error)
 type envVersionUpdater func(*state.Model, version.Number) error
 
-func checkToolsAvailability(ctx context.Context, newEnviron newEnvironFunc, modelCfg *config.Config, finder toolsFinder) (version.Number, error) {
-	currentVersion, ok := modelCfg.AgentVersion()
-	if !ok || currentVersion == version.Zero {
+func (api *AgentToolsAPI) checkToolsAvailability(ctx context.Context) (version.Number, error) {
+	currentVersion, err := api.modelAgentService.GetModelAgentVersion(ctx)
+	if err != nil {
+		return version.Zero, errors.Annotate(err, "getting agent version from service")
+	}
+	if currentVersion == version.Zero {
 		return version.Zero, nil
 	}
 
-	env, err := newEnviron()
+	env, err := api.newEnviron()
 	if err != nil {
 		return version.Zero, errors.Annotatef(err, "cannot make model")
```

Comment:
strictly in this case it's an Environ, which we surface as "cloud provider"
so maybe "cannot make cloud provider" or something